src/main/java/Deadline.java

```diff
@@ -0,0 +1,23 @@
+public class Deadline extends Task{
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "\t[D] " + super.toString() + "(" + "by:" + by + ")";
+    }
+    public String getDescription() {
+        return this.description;
+    }
+    public String getBy() {
+        return this.by;
+    }
+    @Override
+    public String stringToSave() {
+        return "D ==> " + getStatusIcon() + " ==> " + getDescription() + " ==> " + getBy();
```

Comment:
maybe you should use constants in order to reduce the need of magic strings ("==>" and etc...)